csi_camera.cpp

```diff
@@ -0,0 +1,367 @@
+#include <iostream>
```

Comment:
The comments and logs throughout this file are helpful! They make the code a lot more readable and easier to reason about.